### PR TITLE
fix(clerk-js): Added padding for ArrowBlockButton to prevent layout shift

### PR DESCRIPTION
**Pull request by octoper**
## Description

Added some padding when the `ArrowBlockButton` is in loading state and has a left icon to prevent layout shift.

fixes JS-723

### Before

https://github.com/clerkinc/javascript/assets/6823226/d801173e-757c-42d8-9d6b-cfefaa357723

### After

https://github.com/clerkinc/javascript/assets/6823226/064d9f7f-26ce-4920-a04f-3176d2e03041

## Checklist

- [X] `npm test` runs as expected.
- [X] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [X] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx [hunk 1]
```diff
@@ -73,14 +73,21 @@ export const ArrowBlockButton = (props: ArrowBlockButtonProps) => {
...
+              },
+          ]}
         >
           {isLoading ? (
             <Spinner
               elementDescriptor={spinnerElementDescriptor}
               elementId={spinnerElementId}
-              sx={{ position: 'absolute' }}
-              size={'sm'}
+              size={'md'}
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx:90]: 🙃 
Can we explore slightly changing the `Spinner` component here to avoid the `isLoading` check? It feels like the icon and the Spinner should have the exact same dimensions so one could be rendered in place of the other one without layout swift

File: .changeset/large-baboons-hope.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/clerk-js': patch
+---
+
+Added some padding when the ArrowBlockButton is in loading state and has a left icon to prevent layout shift.
```
Comment (nikosdouvlis) [inline .changeset/large-baboons-hope.md:5]: Nitpicky comment follows :)

The text here will be included in our changelog with the next production release and it's intended to be shown to Clerk customers. So I would rephrase it as such:

```suggestion
Updated the OAuth buttons in the SignIn and SignUp components to prevent layout shifts while loading.
```

File: packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx (hunk 1 above)
Comment (octoper) [reply packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx:82]: @nikosdouvlis Just changed it and added padding to the `Spinner` component to avoid the `isLoading` check.

**Review by panteliselef — APPROVED** [review]
Seem like the loading spinner gets bigger. Can you confirm with @cedawson  if that's ok ? Otherwise let's merge

**Review by anagstef — APPROVED** [review]

**Comment by panteliselef** [comment]
Shall we merge this ?

**Comment by cedawson** [comment]
This looks good, thanks @octoper

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
